File: EXAMPLES_C/jetgpio_PWM_example.c
/* Usage example of the JETGPIO library
 * Compile with: gcc -Wall -o jetgpio_PWM_example jetgpio_PWM_example.c -ljetgpio
 * Execute with: sudo ./jetgpio_PWM_example
 */

#include <stdio.h>
#include <stdint.h>
#include <stdlib.h>
#include <unistd.h>
#include <jetgpio.h>

int main(int argc, char *argv[]) {
  int Init;

  Init = gpioInitialise();
  if (Init < 0) {
    /* jetgpio initialisation failed */
    printf("Jetgpio initialisation failed. Error code:  %d\n", Init);
    exit(Init);
  }
  else {
    /* jetgpio initialised okay*/
    printf("Jetgpio initialisation OK. Return code:  %d\n", Init);
  }	

  /* Setting up PWM frequency=10kHz @ pin 32 */

  int PWMstat = gpioSetPWMfrequency(32, 10000);

  if (PWMstat < 0) {
    /* PWM frequency set up failed */
    printf("PWM frequency set up failed. Error code:  %d\n", PWMstat);
    exit(Init);
  }
  else {
    /* PWM frequency set up okay*/
    printf("PWM frequency set up okay at pin 32. Return code:  %d\n", PWMstat);
  }
  /* Set up PWM duty cycle to approx 50% (0=0% to 256=100%) @ pin 32*/

  int PWMstat2 = gpioPWM(32, 128);

  if (PWMstat2 < 0) {
    /* PWM start on failed */
    printf("PWM start failed. Error code:  %d\n", PWMstat2);
    exit(Init);
  }
  else {
    /* PWM started on okay*/
    printf("PWM started up okay at pin 32. Return code:  %d\n", PWMstat2);
  }

  int x =0;
  printf("PWM going at pin 32 for 60 seconds\n");
  while (x<30) {
    sleep(2);
    x++;
  }
  // Terminating library 
  gpioTerminate();
  printf("PWM stopped, bye!\n");
  exit(0);
}

File: github_bad/EXAMPLES_Zig/jetgpio_PWM_example.zig
// Usage example of the JETGPIO library
// Compile with: zig build-exe jetgpio_PWM_example.zig -ljetgpio -lc
// Execute with: sudo ./jetgpio_PWM_example

const std = @import("std");
const jetgpio = @cImport({
    @cInclude("jetgpio.h");
});

pub fn main() !void {
    const init = jetgpio.gpioInitialise();

    if (init < 0) {
        std.debug.print("Error initiating jetgpio: {}\n", .{init});
        std.process.exit(1);
    }

    // Setting up PWM frequency=10kHz @ pin 32

    const PWMstat = jetgpio.gpioSetPWMfrequency(32, 10000);
    if (PWMstat < 0) {
        std.debug.print("PWM frequency set up failed. Error code:  {}\n", .{PWMstat});
        std.process.exit(1);
    }

    // Set up PWM duty cycle to approx 50% (0=0% to 256=100%) @ pin 32

    const PWMstat2 = jetgpio.gpioPWM(32, 128);

    if (PWMstat2 < 0) {
        std.debug.print("PWM start failed. Error code:  {}\n", .{PWMstat2});
        std.process.exit(1);
    }

    var x: i32 = 0;
    std.debug.print("PWM going at pin 32 for 60 seconds\n", .{});

    while (x < 30) {
        std.time.sleep(2000000000);
        x += 1;
    }

    // Terminating library
    jetgpio.gpioTerminate();
}
